datasets/code_search_net/code_search_net.py

```diff
@@ -0,0 +1,216 @@
+    ]
+
+    DEFAULT_CONFIG_NAME = "all"
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "repository_name": datasets.Value("string"),
+                    "func_path_in_repository": datasets.Value("string"),
+                    "func_name": datasets.Value("string"),
+                    "whole_func_string": datasets.Value("string"),
+                    "language": datasets.Value("string"),
+                    "func_code_string": datasets.Value("string"),
+                    "func_code_tokens": datasets.Sequence(datasets.Value("string")),
+                    "func_documentation_string": datasets.Value("string"),
+                    "func_documentation_tokens": datasets.Sequence(datasets.Value("string")),
+                    "split_name": datasets.Value("string"),
+                    "func_code_url": datasets.Value("string"),
+                    # TODO - add licensing info in the examples
+                }
+            ),
+            # No default supervised keys
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators.
+
+        Note: The original data is stored in S3, and follows this unusual directory structure:
+            ```
+            .
+            ├── <language_name>  # e.g. python
+            │   └── final
+            │       └── jsonl
+            │           ├── test
+            │           │   └── <language_name>_test_0.jsonl.gz
+            │           ├── train
+            │           │   ├── <language_name>_train_0.jsonl.gz
+            │           │   ├── <language_name>_train_1.jsonl.gz
+            │           │   ├── ...
+            │           │   └── <language_name>_train_n.jsonl.gz
+            │           └── valid
+            │               └── <language_name>_valid_0.jsonl.gz
+            ├── <language_name>_dedupe_definitions_v2.pkl
+            └── <language_name>_licenses.pkl
+            ```
+        """
+        data_urls = _URLs[self.config.name]
+        # Extracts the language archives
+        lang_names_and_dirs = [(language, dl_manager.download_and_extract(url)) for language, url in data_urls]
+        data_dirs = [
+            os.path.join(lang_dir, lang_name, "final", "jsonl") for lang_name, lang_dir in lang_names_and_dirs
+        ]
+        split_dirs = {
+            split_name: [os.path.join(data_dir, split_name) for data_dir in data_dirs]
+            for split_name in ["train", "test", "valid"]
+        }
+
+        split_paths = {
+            split_name: [
+                dl_manager.extract(entry.path)
+                for split_dir in split_dirs[split_name]
+                for entry in os.scandir(split_dir)
+                if entry.is_file() and entry.name.endswith(".jsonl.gz")
```

Comment:
don't check for the name of the downloaded resources paths, they might be renamed depending on the dl_manager's behavoir